### PR TITLE
refactor txpool options

### PR DESCRIPTION
**Pull request by gtebrean**
## PR description
Add all tx-pool CLI options under a single group https://github.com/hyperledger/besu/issues/5466

## Fixed Issue(s)
Fixes, #5466 

Result
Screenshot 2023-05-30 at 19 40 37

### CODE REVIEW

**Comment by gtebrean** [comment]
I moved the 4 tx-pool stable options from BesuCommand in TransactionPoolOptions in order to can have them in single group.
I tried also to add them in a separated specific TransactionPool group but as TransactionPoolOptions class contains unstable options it didn't work in loading them properly from both stable and unstable options.

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 1]
```diff
@@ -1197,39 +1188,6 @@ static class PermissionsOptionGroup {
           "Sets target gas limit per block. If set, each block's gas limit will approach this setting over time if the current gas limit is different.")
   private final Long targetGasLimit = null;
 
-  // Tx Pool Option Group
-  @CommandLine.ArgGroup(validate = false, heading = "@|bold Tx Pool Options|@%n")
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:1201]: I'd like to keep this grouping for ux. I get your point about the unstable options - can we keep the unstable options on the TransactionPoolOptions class (they don't appear in the help anyway) and put the remaining tx-pool options within this group in BesuCommand

**Review by macfarla — CHANGES_REQUESTED** [review]

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java (hunk 1 above)
Comment (gtebrean) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java]: updated

**Review by macfarla — APPROVED** [review]
LGTM

**Pull request merged**